### PR TITLE
Chao/add vcpu resize support

### PR DESCRIPTION
**Pull request by studychao**

### CODE REVIEW

File: src/api_server.rs [hunk 1]
```diff
@@ -73,7 +74,10 @@ impl ApiServer {
 
         match v["action"].as_str() {
             Some("resize_vcpu") => {
-                println!("mock resize vcpu (will add the actual one when vcpu resize is merged in Dragonball under Kata Container Project");
+                let resize_vcpu_cfg = VcpuResizeInfo {
+                    vcpu_count: v["vcpu_count"].as_u64().map(|count| count as u8),
+                };
+                return self.resize_vcpu(resize_vcpu_cfg);
```
Comment (wllenyj) [inline src/api_server.rs:80]: Do some validation on vcpu_count?

File: src/cli_instance.rs [hunk 2]
```diff
@@ -94,13 +99,17 @@ impl CliInstance {
...
         };
 
-        // check the existence of the serial path (rm it if exist)
-        let serial_file = Path::new(&args.create_args.serial_path);
-        if args.create_args.serial_path != *"stdio" && serial_file.exists() {
-            std::fs::remove_file(serial_file).unwrap();
+        if serial_path.is_some() {
+            // check the existence of the serial path (rm it if exist)
+            // unwrap is safe  because we have check it is Some above.
+            let com1_sock_path = serial_path.unwrap();
```
Comment (HerlinCoder) [inline src/cli_instance.rs]: what about use if let Some(com1_sock_path)

File: src/cli_instance.rs [hunk 3]
```diff
@@ -94,13 +99,17 @@ impl CliInstance {
...
+            let serial_file = Path::new(com1_sock_path.as_str());
+            if serial_file.exists() {
+                std::fs::remove_file(serial_file).unwrap();
```
Comment (HerlinCoder) [inline src/cli_instance.rs]: is unwrap always safe?

File: src/api_server.rs (hunk 1 above)
Comment (studychao) [reply src/api_server.rs:80]: The validation will be in Dragonball code.

File: src/cli_instance.rs (hunk 2 above)
Comment (studychao) [reply src/cli_instance.rs]: fixed

File: src/cli_instance.rs (hunk 3 above)
Comment (studychao) [reply src/cli_instance.rs]: error return added.

**Review by HerlinCoder — APPROVED** [review]

**Review by wllenyj — APPROVED** [review]

**Pull request merged**